test/common/http/utility_test.cc

```diff
@@ -158,6 +158,32 @@ TEST(HttpUtility, getLastAddressFromXFF) {
     EXPECT_EQ(nullptr, ret.address_);
     EXPECT_FALSE(ret.single_address_);
   }
+  {
+    const std::string first_address = "192.0.2.10";
+    const std::string second_address = "192.0.2.1";
+    const std::string third_address = "10.0.0.1";
+    TestHeaderMapImpl request_headers{{"x-forwarded-for", "192.0.2.10, 192.0.2.1 , 10.0.0.1"}};
+
+    // Exercise ltrim().
+    auto ret = Utility::getLastAddressFromXFF(request_headers);
+    EXPECT_EQ(third_address, ret.address_->ip()->addressAsString());
+    EXPECT_FALSE(ret.single_address_);
+
+    // Exercise rtrim() and ltrim().
+    ret = Utility::getLastAddressFromXFF(request_headers, 1);
+    EXPECT_EQ(second_address, ret.address_->ip()->addressAsString());
+    EXPECT_FALSE(ret.single_address_);
+
+    // No space trimming.
+    ret = Utility::getLastAddressFromXFF(request_headers, 2);
+    EXPECT_EQ(first_address, ret.address_->ip()->addressAsString());
```

Comment:
I *think* this test would have passed with the previous PR, because we assumed "X, Y"  The problem is X,Y with no whitespace on either side.

Can we add one more address in this test with no whitespace before the comma on either side?